geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/util/GeoMesaBatchWriterConfig.scala

```diff
@@ -24,6 +24,42 @@ object GeoMesaBatchWriterConfig extends Logging {
       num <- Try(java.lang.Long.parseLong(p)).toOption
     } yield num
 
+  protected [util] def fetchMemoryProperty(prop: String): Option[Long] =
+    for {
+      p <- Option(System.getProperty(prop))
+      num <- parseMemoryProperty(p)
+    } yield num
+
+  protected [util] def parseMemoryProperty(prop: String): Option[Long] = {
+
+    //Scala regex matches the whole string, the leading ^ and trailing $ is implied.
+    val matchSuffix = """(\d+)([A-Za-z]+)?""".r
```

Comment:
The regex """(\d+)([GgKkMm])""".r is a little sharper.  We only need to match our 3 letters, and we only want one copy.